v1/requests.go

```diff
@@ -0,0 +1,51 @@
+package googlemessaging
+
+type FcmMessageBody struct {
+	Message *FcmHttpMessage `json:"message"`
+}
+
+type FcmHttpMessage struct {
+	Topic        string            `json:"topic,omitempty"`
+	Token        string            `json:"token,omitempty"`
+	Notification *FcmNotification  `json:"notification,omitempty"`
+	Data         FcmData           `json:"data,omitempty"`
+	Android      *FcmAndroidConfig `json:"android"`
+}
+
+type FcmNotification struct {
+	Title string `json:"title"`
+	Body  string `json:"body"`
+	Image string `json:"image"`
+}
+
+type FcmAndroidConfig struct {
+	Notification *FcmAndroidNotification `json:"notification"`
+}
+
+type FcmAndroidNotification struct {
+	Icon              string   `json:"icon,omitempty"`
+	Sound             string   `json:"sound,omitempty"`
+	Badge             string   `json:"badge,omitempty"`
+	Tag               string   `json:"tag,omitempty"`
+	Color             string   `json:"color,omitempty"`
+	ClickAction       string   `json:"click_action,omitempty"`
+	ChannelId         string   `json:"channel_id,omitempty"`
+	BodyLocKey        string   `json:"body_loc_key,omitempty"`
+	BodyLocArgs       []string `json:"body_loc_args,omitempty"`
+	TitleLocArgs      []string `json:"title_loc_args,omitempty"`
+	TitleLocKey       string   `json:"title_loc_key,omitempty"`
+	Ticker            string   `json:"ticker,omitempty"`
+	Sticky            string   `json:"sticky,omitempty"`
+	LocalOnly         bool     `json:"local_only,omitempty"`
+	NotificationCount int      `json:"notification_count,omitempty"`
+}
+
+type FcmData map[string]interface{}
```

Comment:
"data" should be string to string map. Maybe in the past it was allowed to have nested objects but the standard is string to string.